src/apps/activities/domain/conditions.py

```diff
@@ -38,19 +86,177 @@ class SliderConditionType(str, Enum):
     OUTSIDE_OF = "OUTSIDE_OF"
 
 
-class OptionPayload(PublicModel):
+class TimePayloadType(str, Enum):
+    START_TIME = "startTime"
+    END_TIME = "endTime"
+
+
+OPTION_BASED_CONDITIONS = [
+    MultiSelectConditionType.INCLUDES_OPTION,
+    MultiSelectConditionType.NOT_INCLUDES_OPTION,
+    SingleSelectConditionType.EQUAL_TO_OPTION,
+    SingleSelectConditionType.NOT_EQUAL_TO_OPTION,
+    SingleSelectionPerRowConditionType.EQUAL_TO_ROW_OPTION,
+    SingleSelectionPerRowConditionType.NOT_EQUAL_TO_ROW_OPTION,
+    MultiSelectionsPerRowConditionType.INCLUDES_ROW_OPTION,
+    MultiSelectionsPerRowConditionType.NOT_INCLUDES_ROW_OPTION,
+]
+
+
+class OptionPayload(PublicModelNoExtra):
     option_value: str
 
 
-class ValuePayload(PublicModel):
+class OptionIndexPayload(OptionPayload):
+    row_index: str
+
+
+class ValuePayload(PublicModelNoExtra):
     value: float
 
     @validator("value")
     def validate_score(cls, value):
         return round(value, 2)
 
 
-class MinMaxPayload(PublicModel):
+class ValueIndexPayload(ValuePayload):
+    row_index: str
+
+
+class SingleDatePayload(PublicModel):
+    date: datetime.date
+
+    def dict(self, *args, **kwargs):
+        d = super().dict(*args, **kwargs)
+        d["date"] = self.date.isoformat()
+        return d
+
+
+class DateRangePayload(PublicModel):
+    minDate: datetime.date
+    maxDate: datetime.date
+
+    @root_validator(pre=True)
+    def validate_dates(cls, values):
+        min_date = values.get("minDate")
+        max_date = values.get("maxDate")
+        if min_date and max_date and min_date > max_date:
+            raise ValueError("minDate cannot be later than maxDate")
+        return values
+
+    def dict(self, *args, **kwargs):
+        d = super().dict(*args, **kwargs)
+        d["minDate"] = self.minDate.isoformat()
+        d["maxDate"] = self.maxDate.isoformat()
+        return d
+
+
+class TimePayload(PublicModel):
+    type: str | None = None
+    value: datetime.time
+
+    def dict(self, *args, **kwargs):
+        d = super().dict(*args, **kwargs)
+        d["value"] = self.value.strftime("%H:%M")
+        return d
+
+
+class SingleTimePayload(PublicModel):
```

Comment:
The Time and TimeRange types appear to be using 2 different formats and both the Admin and Web apps are unable to parse the `{ time: "00:00" }` format properly. The `{housr: 0, minutes: 0}` format appears to work correctly

Payload
![CleanShot 2024-10-02 at 20 11 14](https://github.com/user-attachments/assets/a850dd01-f951-4256-938b-02c0591a4959)

Admin Error
![CleanShot 2024-10-02 at 20 11 37](https://github.com/user-attachments/assets/1de9347a-c630-4d1a-956b-a489cbbe6d07)

Web app crashes
![CleanShot 2024-10-02 at 20 14 44](https://github.com/user-attachments/assets/4b8087bc-7137-4d3a-859b-2df9956fa424)

Reply:
Note that this issue doesn't appear to break Date type, I tested greater, equal, less, between, outside and all worked OK for Date.